### PR TITLE
Take baseos activation status from zboot

### PR DESCRIPTION
**Pull request by europaul**
It is safer to get the activation status of the baseOS from zboot instead of the pubsub, since pubsub only gets populated after a config is read. This information is needed to determine whether the app-related part of the config should be inhaled. In particular consider the following scenario:
- a new config arrives from the controller and has a baseos.Activate field set to true, however this baseos version is already activated
- EVE checks whether it's a known baseos and since pubsub is empty after a reboot - EVE decides that it is
- EVE skips inhaling the config, because it thinks it should activate a new baseos, however the activation is already done - so nothing happens

This PR is supposed to help mitigate such a scenario.

### CODE REVIEW

**Comment by eriknordmark** [comment]
Regarding this case:
> a new config arrives from the controller and has a baseos.Activate field set to true, however this baseos version is already activated
there is a sub-case when the device has been installed with EVE version X.Y.Z (from PXE or USB) but the controller has not yet been setup to specify the version of EVE for this device.
If you (or some automation) then tells the controller that the EVE version should be X.Y.Z, then you will end up in the same "skip" state. In this case there is no previous config (from a checkpoint or otherwise) which you can use.

File: pkg/pillar/cmd/zedagent/zedagent.go [hunk 1]
```diff
@@ -1156,6 +1156,7 @@ func initPublications(zedagentCtx *zedagentContext) {
 		log.Fatal(err)
 	}
 	getconfigCtx.pubBaseOsConfig.ClearRestarted()
+	populateBaseOsConfig(getconfigCtx)
```
Comment (rouming) [inline pkg/pillar/cmd/zedagent/zedagent.go:1159]: In general I do not see any problems in inhaling config earlier (here I can be mistaken). But calling `populateBaseOsConfig()` from the `initPublications()` is misleading: you do not expect that a config will be read and something published on that path. 

As a suggestion: why not to extend the `maybeLoadBootstrapConfig()` (which happens a few lines after the `initPublications` is called) and inhale either whole bootstrap or whole last config ?

**Comment by europaul** [comment]
> Regarding this case:
> 
> > a new config arrives from the controller and has a baseos.Activate field set to true, however this baseos version is already activated
> 
> there is a sub-case when the device has been installed with EVE version X.Y.Z (from PXE or USB) but the controller has not yet been setup to specify the version of EVE for this device. If you (or some automation) then tells the controller that the EVE version should be X.Y.Z, then you will end up in the same "skip" state. In this case there is no previous config (from a checkpoint or otherwise) which you can use.

I see :+1: I think I've seen the same behavior when starting `make run` and then getting a config with the same baseos.

File: pkg/pillar/cmd/zedagent/zedagent.go (hunk 1 above)
Comment (europaul) [reply pkg/pillar/cmd/zedagent/zedagent.go:1159]: Noted :+1: That makes sense, but I will have to redo the PR given the reason in @eriknordmark's comment

**Comment by europaul** [comment]
@eriknordmark @rouming @OhmSpectator I reworked the PR based on @eriknordmark's comment, please have another look.

**Comment by OhmSpectator** [comment]
Was this approach tested with a simple EVE upgrade test?
As far as I remember we have one in Eden testset (not sure).

**Comment by europaul** [comment]
> Was this approach tested with a simple EVE upgrade test? As far as I remember we have one in Eden testset (not sure).

I tested the upgrade manually. The upgrade tests are indeed included in Eden and will run once the PR is approved.

**Review by eriknordmark — APPROVED** [review]
LGTM

**Comment by milan-zededa** [comment]
@eriknordmark @shjala @rouming 
One of the upgrade tests failed with:
```
Reboot from agent verifier[1380] in partition IMGB at EVE version 9.7.0-kvm-amd64 at 2024-03-14T22:42:09.48986295Z: fatal: agent verifier[1380]: mkdir /persist/vault/verifier: required key not available
```
I don't know if it is something worth investigating further. Maybe related to fscrypt changes?
But note that the upgrade test is also using the unsupported downgrade to first get from the PR version of EVE to 9.7, then upgrading back to the PR version. We should change it to start with a fresh 9.7 (or another older version), then upgrade to the PR version (now that upgrade is a separate workflow, it should be relatively easy; CC @uncleDecart ).

**Comment by europaul** [comment]
@milan-zededa thank you for digging deeper into the failing test! If somebody is going to debug this further, keep in mind that it only happened on ext4, the same tests runs fine for zfs.

**Comment by eriknordmark** [comment]
The Eden tests which downgrade will hit this now. The new fscrypt from @rouming uses v2 policies which will not be recognized after downgrading. So the suggesting to have eden install old, update to new, then update back to old, is what we need to test.
[It is just odd that the verifier fails; I'd expect it to fail to unlock the vault.]

**Pull request merged**